framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -795,7 +795,14 @@ void MediaPlayerImpl::playback()
 			}
 		}
 	} else if (num_read == 0) {
-		stop();
+		audio_manager_result_t result = stop_audio_stream_out();
+		if (result != AUDIO_MANAGER_SUCCESS) {
+			meddbg("stop_audio_stream_out failed ret : %d\n", result);
```

Comment:
1. It means that h/w operation failed, it should be sent to user.
2. modified code is already used in stopPlayer, you can make function for this code. ex. stopPlayback

Reply:
1. There is no appropriate error callback for finish error. Should we add `PLAYER_OBSERVER_COMMAND_FINISH_ERROR`?
2.`stopPlayback` called `notifyObserver(PLAYER_OBSERVER_COMMAND_STOPPED)`.
User don't want to get `PLAYER_OBSERVER_COMMAND_STOPPED` and `PLAYER_OBSERVER_COMMAND_FINISHIED` together.
That's why we modify the logic.

Reply:
PLAYER_OBSERVER_COMMAND_PLAYBACK_ERROR is acceptable but your suggestion also good for me.